cmd/postgres_exporter/postgres_exporter.go

```diff
@@ -1497,10 +1496,9 @@ func (e *Exporter) scrape(ch chan<- prometheus.Metric) {
 func (e *Exporter) discoverDatabaseDSNs() []string {
 	dsns := make(map[string]struct{})
 	for _, dsn := range e.dsn {
-		parsedDSN, err := url.Parse(dsn)
+		parsedDSN, err := pq.ParseURL(dsn)
```

Comment:
What sort of errors can parseurl throw? Would just using the DSN on any error cause unforeseen issues further along in code?